bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/internal/GC100IRBinding.java

```diff
@@ -0,0 +1,262 @@
+        }
+
+        GC100IRHost gc100Host;
+        if (gc100InstanceName.startsWith("#")) {
+            // trim off the '#' identifier
+            String instance = gc100InstanceName.substring(1);
+
+            // check if we have been initialised yet - can't process
+            // named instances until we have read the binding config
+            if (nameHostMapper == null) {
+                logger.trace("Attempting to access the named instance '{}' before the binding config has been loaded",
+                        instance);
+                return null;
+            }
+
+            // check this instance name exists in our config
+            if (!nameHostMapper.containsKey(instance)) {
+                logger.error("Named instance '{}' does not exist in the binding config", instance);
+                return null;
+            }
+
+            gc100Host = nameHostMapper.get(instance);
+        } else {
+            gc100Host = new GC100IRHost();
+            gc100Host.setHostname(gc100InstanceName);
+        }
+
+        // create a new connection handler
+        logger.debug("Creating new GC100IRConnector for '{}' on {}", gc100InstanceName, gc100Host.getHostname());
+        connector = new GC100IRConnector(gc100Host);
+        connectors.put(gc100InstanceName, connector);
+
+        // attempt to open the connection straight away
+        try {
+            connector.open();
+        } catch (Exception e) {
+            logger.error("Connection failed for '{}' on {}", gc100InstanceName, gc100Host.getHostname());
+        }
+
+        return connector;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void internalReceiveCommand(String itemName, Command command) {
+        try {
+            // lookup the GC100 instance name and property for this item
+            String gc100Instance = getGC100InstanceName(itemName);
+            GC100IRConnector connector = getGC100Connector(gc100Instance);
+
+            if (connector == null) {
+                logger.warn("Received command ({}) for item {} but no GC100 connector found for {}, ignoring",
+                        command.toString(), itemName, gc100Instance);
+                return;
+            }
+
+            if (!connector.isConnected()) {
+                logger.warn(
+                        "Received command ({}) for item {} but the connection to the GC100 instance {} is down, ignoring",
+                        command.toString(), itemName, gc100Instance);
+                return;
+            }
+            connector.invokeCommand(itemName);
+
+        } catch (Exception e) {
+            logger.error("Error handling command", e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+        Map<String, GC100IRHost> hosts = new HashMap<String, GC100IRHost>();
+        Enumeration<String> keys = config.keys();
```

Comment:
I think it's possible for this method to be called with a `null` for the `config` parameter.  Other bindings check for `null` before using it.  Sorry I didn't notice this earlier.